test/python/transpiler/test_basis_translator.py

```diff
@@ -854,4 +854,4 @@ def test_skip_target_basis_equivalences_1(self):
             routing_method="sabre",
             seed_transpiler=42,
         )
-        self.assertEqual(circ_transpiled.count_ops(), {"cx": 91, "rz": 66, "sx": 22})
+        self.assertEqual(circ_transpiled.count_ops(), {"cx": 100, "rz": 66, "sx": 22})
```

Comment:
I agree with your comment in the commit message, this test is poorly written. This was originally added to verify a fix for a very specific basis translation issue with unnecessary translating of gates already in the basis, but it's too broad. I hit the same exact issue in #6302 and rewrote the test to be more targeted there so it's just about the basis translator and not implicitly encoding sabre behavior. I'll extract that from the PR (because that PR is blocked on a retworkx PR and release) as a standalone so we neither #6302 or this will have to worry about the test.